lib/nightwatch-mocks-handler.js

```diff
@@ -33,8 +42,32 @@ _.extend(MocksHandler.prototype, {
    * or started manually using start/stop async callbacks in the format
    * { mockServer: { start: func({port}), stop: ... } }
    */
+
   before: function (options, callback) {
     var log = this.log;
+    var startServer = function (server, connectOptions, serverPlugin) {
+
+      var deferred = Q.defer();
+      server.connection(connectOptions);
+      server.register(serverPlugin, function (err) {
+        if (err) {
+          log('Error in registering');
+          log(err);
+          deferred.reject(new Error(err));
```

Comment:
if we have a failure we don't want to start the server.  should add a `return` statement here